File: Python/1. Two Sum.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        maple = dict()
        ans = list()
        for i in range(len(nums)):
            tmp = target - nums[i]
            if maple.get(tmp) != None:
                ans.append(maple[tmp])
                ans.append(i)
            maple[nums[i]] = i
        return ans

File: C++/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> maple;
        vector<int> ans;
        int len = nums.size();
        for(int i=0;i<len;i++)
        {
            int tmp = target - nums[i];
            if(maple.find(tmp) != maple.end())
            {
                ans.push_back(maple[tmp]);
                ans.push_back(i);
            }
            maple[nums[i]] = i;
        }
        return ans;
    }
};